src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -632,12 +633,25 @@ ReactDOMComponent.Mixin = {
           this._currentElement.type
         );
       }
+      var isCustomComponentTag = isCustomComponent(this._tag, props);
+      if (__DEV__ && isCustomComponentTag && !didWarnShadyDOM && el.shadyRoot) {
+        var owner = this._currentElement._owner;
+        var name = owner.getName() || 'A component';
```

Comment:
`owner` might be missing so we should check that it's there first.
`owner && owner.getName() || 'A component` should probably work, can you check other similar places?